integtest/src/main/java/org/corfudb/universe/cluster/docker/FakeDns.java

```diff
@@ -0,0 +1,180 @@
+     */
+    private boolean installed = false;
+
+    private FakeDns() {
+    }
+
+    public static FakeDns getInstance() {
+        return instance;
+    }
+
+    public synchronized void addForwardResolution(String hostname, InetAddress ip) {
+        forwardResolutions.put(hostname, ip);
+    }
+
+    public synchronized void addReverseResolution(InetAddress ip, String hostname) {
+        reverseResolutions.put(ip, hostname);
+    }
+
+    /**
+     * Install the fake DNS resolver into the Java runtime.
+     */
+    public synchronized FakeDns install() {
+        if (installed) return this;
+        try {
+            try {
+                // Override the NameService in Java 9 or later.
+                final Class<?> nameServiceInterface = Class.forName("java.net.InetAddress$NameService");
+                Field field = InetAddress.class.getDeclaredField("nameService");
+                // Get the default NameService to fallback to.
+                Method method = InetAddress.class.getDeclaredMethod("createNameService");
+                method.setAccessible(true);
+                Object fallbackNameService = method.invoke(null);
+                // Create a proxy instance to set on the InetAddress field which will handle
+                // all NameService calls.
+                Object proxy = Proxy.newProxyInstance(
+                        nameServiceInterface.getClassLoader(),
+                        new Class<?>[]{nameServiceInterface},
+                        new NameServiceListener(fallbackNameService)
+                );
+                field.setAccessible(true);
+                field.set(InetAddress.class, proxy);
+            } catch (final ClassNotFoundException | NoSuchFieldException e) {
+                // Override the NameService in Java 8 or earlier.
+                final Class<?> nameServiceInterface = Class.forName("sun.net.spi.nameservice.NameService");
+                Field field = InetAddress.class.getDeclaredField("nameServices");
+                // Get the default NameService to fallback to.
+                Method method = InetAddress.class.getDeclaredMethod("createNSProvider", String.class);
+                method.setAccessible(true);
+                Object fallbackNameService = method.invoke(null, "default");
+                // Create a proxy instance to set on the InetAddress field which will handle
+                // all NameService calls.
+                Object proxy = Proxy.newProxyInstance(
+                        nameServiceInterface.getClassLoader(),
+                        new Class<?>[]{nameServiceInterface},
+                        new NameServiceListener(fallbackNameService)
+                );
+                field.setAccessible(true);
+                // Java 8 or earlier takes a list of NameServices
+                field.set(InetAddress.class, Arrays.asList(proxy));
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)